standards/application/messaging-api.md

```diff
@@ -0,0 +1,965 @@
+The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, 
+“RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in [RFC2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+### Definitions
+
+This section defines key terms and concepts used throughout this document.
+Each term is detailed in dedicated subsections to ensure clarity and consistency in interpretation.
+
+`Multiaddr` - a self-describing format for encoding network address of a remote peer as per [libp2p addressing](https://github.com/libp2p/specs/blob/6d38f88f7b2d16b0e4489298bcd0737a6d704f7e/addressing/README.md) specification.
+
+### Motivation
+
+Real-world application development has exposed challenges in achieving comprehensive usability guarantees for core Waku protocols.
+Although recommendations such as [P2P-RELIABILITY](./p2p-reliability.md) have been introduced to enhance protocol robustness,
+the fragmented landscape of guidelines and the requirement to re-implement reliability strategies across diverse use cases remain evident.
+
+A significant challenge has been the lack of a unified abstraction for message exchange.
+Developers require a straightforward mechanism to send,
+receive, and track messages—whether originating from a node or propagated throughout its connected network.
+
+## API design
+
+### Requirements
+
+This API is designed for generic use and ease of implementation across `nwaku`, `js-waku`,
+and traditional `REST` architectures.
+From this point forward,
+relevant code blocks will be provided in `JSON` format, `HTTP` format or by using `TypeScript` syntax.
+
+### Initial configuration
+
+The `Messaging API` is an abstraction layer built upon the basic functionality provided by a node.  
+The configuration definitions that follow omit implementation-specific details,
+focusing exclusively on settings explicitly required by the `Messaging API`.
+
+```typescript
+{
+  mode: "edge" | "service";
+  clusterId: number;
+  shards: number[];
+  storeNodes?: string[];
+  serviceNodes?: string[];
+  bootstrapNodes: string[];
+  confirmStored?: boolean;
+  confirmReceived?: boolean;
+  confirmContentTopics?: string[];
+}
+```
+
+#### Properties
+
+##### `mode`
+This property defines behavior of the node and MUST be specified.
+
+If `edge` selected the node MUST use [LIGHTPUSH](../standards/core/lightpush.md) for sending messages,
+and [FILTER](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md) for receiving messages.
+
+If `service` selected the node MUST implement [RELAY](https://github.com/vacp2p/rfc-index/blob/0277fd0c4dbd907dfb2f0c28b6cde94a335e1fae/waku/standards/core/11/relay.md),
+[STORE](../standards/core/store.md) as well as host endpoint for [LIGHTPUSH](../standards/core/lightpush.md) and [FILTER](https://github.com/vacp2p/rfc-index/blob/7b443c1aab627894e3f22f5adfbb93f4c4eac4f6/waku/standards/core/12/filter.md).
```

Comment:
Requiring store does not feel right - Store nodes have different expectations than Lightpush and Filter nodes - those can come and go and still be useful (unless the fluctuation is in order of seconds or minutes), but Store (even with Store sync) should be more stable and available + enabling the protocol puts requirement on hosting the DB. 

I'd personally see store managed separately

Reply:
To add to it, I might be fine if my phone is a `service` node when on Wifi and plugged in the charger, but I would not want it to store GBs of random and "useless" data:)